torch/lib/c10d/test/ProcessGroupGlooTest.cpp

```diff
@@ -203,11 +202,10 @@ void testBroadcast(const std::string& path, const at::Backend b) {
         inputs[k].resize(stride);
         at::DeviceGuard deviceGuard;
         for (auto l = 0; l < stride; l++) {
-          if (type.is_cuda()) {
+          if (b == at::Backend::CUDA) { // NB:wouldn't work with sparse
```

Comment:
I pessimized this site because there wasn't a convenient way to check if a Backend was a CUDA backend, and since Backend is on its way out anyway, I am hoping when that happens someone can just rewrite this code to not be problematic anymore.

In the test file, Backend is only ever passed `CUDA` and `CPU`.  I think it's safe to assume that if someone is adding a test for `SparseCUDA`, they will have to substantially rewrite this test file anyway. Also, you would never actually want to create a sparse tensor full of ones, unless we're actually going to implement that "sparse tensor + scalar" idea, which I don't think we were planning to?

Reply:
I would probably write this test differently, but it is not bad enough that it warrants refactoring.